### ISSUE TITLE
Add remaining filters excluded for elasticsearch

### ISSUE THREAD

**Issue by samschlicht** [talentbeyondboundaries/tbbtalentv2#678]
For #454 and #677 to function properly — and generally to bring elasticsearch into alignment with regular — we need to bring elasticsearch capability to the remaining filters that don't have it:
- [x] Min age
- [x] Max age
- [x] Min years experience
- [x] Max years experience
- [x] Mini intake completed?
- [x] Full intake completed?
- [x] Last Modified

**Comment by samschlicht** [comment]
Noting that there are also the extra intake fields now (and check for any others)

**Comment by samschlicht** [comment]
Age is done but we'll need to add the following fields to the elasticsearch index:

- candidate > mini_intake
- candidate > full_intake
- candidate_occupation > years_experience

Once all the fields are available (and kept that way - regular and elastic search need to be aligned) #677 will be fairly simple — in my view, we just start to process all joined searches as elastic.

**Comment by samschlicht** [comment]
Note to self that if this issue isn't cleared for next release, we need to make the intake filter read-only when search is elastic — at the moment they're available but don't do anything.

**Comment by samschlicht** [comment]
Found another field!
- [ ] Modified between

**Comment by samschlicht** [comment]
Tested?
 - [x] Min age
 - [x] Max age
 - [x] Min years experience
 - [x] Max years experience
 - [x] Mini intake completed?
 - [x] Full intake completed?
 - [x] Last Modified

**Comment by samschlicht** [comment]
We've parked this PR while #871 is resolved

**Comment by samschlicht** [comment]
This was at PR stage but needs to be revised in light of recent changes to ES mapping.

Next steps:
- Discuss w Sadat and make required changes
- While I'm at it, perhaps included UNHCR status field which has recently been added to regular search (#928)

**Comment by samschlicht** [comment]
Discussed today w Sadat, confirming:
- [x] add UNHCR status to the CandidateEs java class and ES search processing
- [x] rename esOccupations unless I can find the compelling reason for not using the generic Occupations
- [x] manual test to make sure occupations still working
- [x] boot up Spring without the new annotation that stops the auto-creation of mappings, using the resulting mapping from Kibana to be copy-pasted into a V3 of the Python script that we're using to create the ES indexes 
- [x] double-check that _all_ the search filters are actually available and working for ES — _yes, except candidate opps_

**Comment by samschlicht** [comment]
Noting for memory that Sadat also needs to add to the mappings and may just want to build on my branch.

**Comment by samschlicht** [comment]
Glad I did a proper check — I've repeated an old mistake and only considered fields that were disabled when a keyword search term was entered. Added:

- [x] Survey Type
- [ ] Candidate job opportunity cases — _will move this forward as a new issue_
(This will be an interesting one: I don't want to start saving and having to update candidate opps in the ES index. I think it'll have to be processed similarly to exclusion list candidates, i.e., results are computed in a separate _regular_ search query, and the resulting set of candidate IDs used to construct an ES filter — see SavedSearchServiceImpl 1025. The good news is I don't have to update the mapping for this one.)

This one is included but not working properly, noting also the bug described in 1027:
- [x] Referrer
